test/connectivity/connectivity.test.jsx

```diff
@@ -0,0 +1,259 @@
+const { expect } = require('chai');
+const Enzyme = require('enzyme');
+const Adapter = require('enzyme-adapter-react-16');
+const jsdomGlobal = require('jsdom-global');
+const m = require('module');
+const React = require('react');
+const debug = require('debug')('connectivity-test');
+const { promisify } = require('util');
+
+Enzyme.configure({ adapter: new Adapter() });
+const { mount } = Enzyme;
+
+// Stub electron module for tests
+const originalLoader = m._load;
+const stubs = {
+  electron: {
+    ipcMain: {
```

Comment:
why is this necessary?

Reply:
I tried using `electron-mocha` to fully mock the electron functions, however the `hadron-ipc` module that `storage-mixin` requires I think depends either on an older version of electron's commands, or requires both the renderer and main process to be running. I spent a bit of time trying to get it work and couldn't quite get it, so I figured this does the job. Totally open to suggestions here, I think it might be nice to keep these tests lightweight so I was avoiding spectron.

These are two of the electron parts we need to mock for reference:
https://github.com/mongodb-js/hadron-ipc/blob/master/lib/renderer.js#L10
https://github.com/mongodb-js/compass-connect/blob/master/src/components/form/form-file-input.jsx#L9

Reply:
Was the issue something like 'the module did not register'?

Reply:
It was cannot access `dialog` of undefined https://github.com/mongodb-js/compass-connect/blob/master/src/components/form/form-file-input.jsx#L9
or cannot call undefined function call https://github.com/mongodb-js/hadron-ipc/blob/master/lib/renderer.js#L10 (which happens when running in the electron main process and I think that makes sense because that's expected to be run in the renderer process)